SimPEG/Utils/io_utils.py

```diff
@@ -146,35 +147,72 @@ def surface2inds(vrtx, trgl, mesh, boundaries=True, internal=True):
     return insideGrid
 
 
-def remoteDownload(url, remoteFiles, basePath=None):
+def remoteDownload(
+    url, remoteFiles, path='.', directory='SimPEGtemp', rm_previous=False
+):
     """
     Function to download all files stored in a cloud directory
-    var:    url ("http:\\...")
-    list:   List of file names to download
+
+    :param str url: url of the storage bucket ("http://...")
+    :param list remoteFiles: List of file names to download from the storate bucket
+    :param str path: path to where the directory is created and files downloaded (default is the current directory)
+    :param str directory: name of the directory to be created and have content downloaded to
+    :param bool rm_previous: remove file and contents if a directory with the specified name already exists
     """
 
     # Download from cloud
     import urllib
     import shutil
     import os
     import sys
+
+    def rename_path(downloadpath):
+        splitfullpath = downloadpath.split(os.path.sep)
+        curdir = splitfullpath[-1]
+        splitdir = curdir.split('(')
+        rootdir = splitdir[0]
+
+        # add (num) to the end of the filename
+        if len(splitdir) == 1:
+            num = 1
+        else:
+            num = int(splitdir[-1][:-1])
+            num += 1
+
+        return os.path.sep.join(
+            splitfullpath[:-1] + [rootdir + '({})'.format(num)]
+        )
+
+    # grab the correct url retriever
     if sys.version_info < (3,):
         urlretrieve = urllib.urlretrieve
     else:
         urlretrieve = urllib.request.urlretrieve
 
-    if basePath is None:
-        basePath = os.curdir+os.path.sep+'SimPEGtemp'+os.path.sep
-
-    if os.path.exists(basePath):
-        shutil.rmtree(basePath)
-
-    os.makedirs(basePath)
-
-    print("Download files from URL...")
+    # ensure we are working with absolute paths
+    path = os.path.abspath(path)
+    downloadpath = os.path.sep.join([path]+[directory])
```

Comment:
Yeah, this is likely confusing. `path` is the path to your working directory, `directory` is the name of the new folder we are adding where we download things. 

Thoughts on alternatives?